store/src/main/java/org/apache/rocketmq/store/dledger/DLedgerCommitLog.java

```diff
@@ -421,6 +421,11 @@ private void setMessageInfo(MessageExtBrokerInner msg, int tranType) {
 
         setMessageInfo(msg, tranType);
 
+        boolean multiDispatchWrapResult = multiDispatch.wrapMultiDispatch(msg);
```

Comment:
multiDispatch.wrapMultiDispatch() has a concurrency problem.

Reply:
fixed

Reply:
> fixed

the keyBuilder can be ThreadLocal?